pkg/k8scontext/context.go

```diff
@@ -218,6 +218,26 @@ func (c *Context) ListPodsByServiceSelector(selector map[string]string) []*v1.Po
 	return podList
 }
 
+// IsPodReferencedByAnyIngress provides whether a POD is useful i.e. a POD is used by an ingress
+func (c *Context) IsPodReferencedByAnyIngress(pod *v1.Pod) bool {
+	// first find all the services
+	services := c.listServicesByPodSelector(pod.Labels)
```

Comment:
I'm typically not too worried about efficiency, but I know that this function would get high QPS since we'd be calling it for each event.  In that case - it might be prudent to pass a reference to the object in lieu of allocating and copying a `map[string]string`, which is the case with `pod.Labels`

What do you think about tweaking this to `services := c.listServicesByPodSelector(pod)`, which changes the semantics of the function a bit - perhaps if you agree to that change - the function could be called `listServicesForPod(pod)` or something similar?